src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -909,6 +909,15 @@ internal int CalculateSyntaxOffsetInSynthesizedConstructor(int position, SyntaxT
                 return syntaxOffset;
             }
 
+            if (declaration.Declarations.Length == 1 && position == declaration.Declarations[0].Location.SourceSpan.Start)
```

Comment:
I misunderstood the role of Declarations here--I thought it was the set of variables being closed over rather than the set of declarations of the type. I'll write a test and make necessary adjustments.